### DISCUSSION TITLE
Block selection of text

### DISCUSSION THREAD

**Discussion by mnussbaum** [ghostty-org/ghostty#3307] [Q&A] [answered]
One feature I find highly useful in a terminal is the ability select a rectangular block of text. On the macOS terminal this can be done by holding `opt` while clicking and dragging the cursor over text. And for another example, the documentation for this feature in wezterm is [here](https://wezfurlong.org/wezterm/config/lua/keyassignment/SelectTextAtMouseCursor.html), where it describes the `SelectTextAtMouseCursor("Block")` action.

Block selection is particularly helpful in the context of a tool like tmux, where a user might want to select multiple lines of text in one pane, without pulling in text from a neighboring horizontal split.

From my initial read through of the Ghostty documentation, I haven't seen any mention of block or rectangular selection. Is it available? If not I think it would be a great feature to add!

**Comment by mitchellh** [comment]
Ghostty supports this. I’m not at a computer to double check the correct modifiers to press but it is supported.

**Reply by mnussbaum** [reply]
That's great to hear! And thanks for the lightning fast response :smile:. Apologies if I've overlooked docs on how it can be triggered

I'm specifically curious about how to trigger the behavior on linux, fwiw.

**Reply by jcollie** [reply, marked as answer]
Hold down `ctrl` and `alt` and the cursor should change to a crosshair cursor. You can then select a rectangular region.

**Reply by mnussbaum** [reply]
Holding down `ctrl` and `alt` does indeed trigger a cross-hairs for me, but then when I click and drag no text is selected. Is there any info I can provide that would be helpful to check if this is a bug or not?

**Reply by mnussbaum** [reply]
After playing around a little, I found that when I enter full screen mode I can indeed block select using `ctrl` and `alt`, which is great! I wonder if part of the reason it isn't working for me outside of fullscreen mode is because `alt` is my desktop environment's mod key, which might be causing some weird interaction?

Let me know if there's any info I can provide to debug the non-fullscreen failure to select. I'd also be fine remapping the action to a different key combo, if that's possible.

Thanks for the pointers so far!

**Comment by jcollie** [comment]
Yes, if your DE is intercepting the `alt` key that would affect this feature. I'm on mobile right now but I don't recall a way to customize this in Ghostty. I'm not sure because I've never needed to change this.

**Reply by mnussbaum** [reply]
Makes sense. If you find there isn't an existing way to remap this key, do you think that would make sense as a feature request? Fwiw `alt` is a pretty common mod key since it's the default for the i3 and Sway DEs on Linux

**Reply by 00-kat** [reply]
@mnussbaum Opened #3848.

**Comment by towolf** [comment]
Would it be possible to allow holding only `ctrl` for this? 

I'm used to this from gnome-terminal and it's very much ingrained in my brain to use this modifier only.

And I'm left-handed and use my touch pad, and holding `ctrl-alt` on the left-and side doesn't work like that.

**Comment by reinierbutot** [comment]
is there a way to use visual selection mode like in vim? i can use the mouse to select text using ctrl+alt but i would like to keep my fingers on the keyboard and not have to reach for the mouse...

**Reply by 00-kat** [reply]
#3488 #3708.

**Comment by rcgoodfellow** [comment]
For those coming here on a mac, the key combo is `option-shift`.
